modules/engines/engine-xsodata/src/main/java/com/sap/xsk/xsodata/ds/filter/XSODataForwardFilter.java

```diff
@@ -28,25 +34,27 @@ public void init(FilterConfig filterConfig) throws ServletException {
   @Override
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
       throws IOException, ServletException {
-    XSKODataWrappedHttpServletRequest wrappedHttpServletRequest = new XSKODataWrappedHttpServletRequest((HttpServletRequest) request);
+    XSKWrappedHttpServletRequest wrappedHttpServletRequest = new XSKWrappedHttpServletRequest((HttpServletRequest) request);
 
     //only on production case
     String editorHeader = wrappedHttpServletRequest.getHeader("Dirigible-Editor");
     String requestMethod = wrappedHttpServletRequest.getMethod();
 
-    if (editorHeader == null && requestMethod.equals("POST")) {
-      wrappedHttpServletRequest.putHeader("Dirigible-Editor", "Workspace");
-    } else if (editorHeader == null) {
+    if (editorHeader == null) {
       String uri = wrappedHttpServletRequest.getRequestURI();
       int index = uri.indexOf(".xsodata");
       if (index > 0) {
-        String parameters = "";
-        if (uri.length() > index + (".xsodata".length() - 1)) {
-          parameters = uri.substring(index + ".xsodata".length());
-        }
+        if (requestMethod.equals("POST")) {
+          wrappedHttpServletRequest.putHeader("Dirigible-Editor", "Workspace");
```

Comment:
@aihanrashidov, I'm not sure, if this won't break the behaviour of the XSOData (OData) services, when called with HTTP **POST** method.

Could you double check that behaviour:
- Create a ***.xsodata** file with one simple entity (table with 1 or 2 columns)
- Insert some data into the database table via the Database perspective
- Call the ***.xsodata** service with a **GET** request to retrieve the data
- Execute a **POST** request with JSON body to insert some data into the XSOData service (via Postman)

In case that the above steps fails, please double check the same scenario, but without the changes coming from the PR (there could be an issue with the XSOData in general).